### PR TITLE
Ilp quoter backend

### PR DESCRIPTION
**Pull request by gip**

### CODE REVIEW

File: src/backends/ilp-quoter/index.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+      destination_ledger: params.destination_ledger,
+      source_amount: result.body.source_amount,
+      destination_amount: result.body.destination_amount
+    }
+    return quote
+  }
+
+  * hasPair (source, destination) {
+    const pair = lookupCurrencies(source, destination)
+    const r = this.pairs.find((p) => (p[0] === pair[0]) && (p[1] === pair[1])) !== undefined
```
Comment (MatthewPhinney) [inline src/backends/ilp-quoter/index.js]: Can you use a more descriptive variable name (instead of `r`)?

File: src/backends/ilp-quoter/index.js [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+const NoAmountSpecifiedError = 
+  require('../../errors/no-amount-specified-error')
+const UnacceptableQuoterAmountError = 
+  require('../../errors/unacceptable-quoter-amount-error')
+const log = require('../../services/log')('ilpquote')
+const config = require('../../services/config')
+
+function lookupCurrencies (source_ledger, destination_ledger) {
+  for (let pair of config.get('tradingPairs').toJS()) {
+    if (pair[0].indexOf(source_ledger) === 4 &&
```
Comment (MatthewPhinney) [inline src/backends/ilp-quoter/index.js]: What is the significance of `4`?

File: src/backends/ilp-quoter/index.js [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+  return null
+}
+
+/**
+ * Example backend that connects to an external component to get
+ *   the source and destination amounts
+ */
+class ILPQuoter {
+  constructor (opts) {
+    log.warn('ctor')
```
Comment (MatthewPhinney) [inline src/backends/ilp-quoter/index.js]: Shouldn't this be a `debug` log?

File: src/backends/ilp-quoter/index.js (hunk 2 above)
Comment (gip) [inline src/backends/ilp-quoter/index.js]: This code was copied from the fixerio backend. I'll look into it.

**Comment by emschwartz** [comment]
Could you include more of a description in the commit for what this functionality is for?

File: src/backends/ilp-quoter/index.js [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+                '?type=' + type
+    const result = yield request({ uri, json: true })
+    const fixedAmount = type === 'source' ? result.body.source_amount
+                                          : result.body.destination_amount
+    if (fixedAmount !== amount.toString()) {
+      throw new UnacceptableQuoterAmountError('Backend returned an invalid ' +
+                type + ' amount: ' + fixedAmount + ' (expected: ' + amount + ')')
+    }
+    const quote = {
+      source_ledger: params.source_ledger,
```
Comment (emschwartz) [inline src/backends/ilp-quoter/index.js]: It should be documented somewhere that the amounts should be queried and returned as strings

File: src/backends/ilp-quoter/index.js [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+      destination_ledger: params.destination_ledger,
+      source_amount: result.body.source_amount,
+      destination_amount: result.body.destination_amount
```
Comment (emschwartz) [inline src/backends/ilp-quoter/index.js]: It should be documented somewhere that the amounts should be queried and returned as strings

Comment (gip) [inline src/backends/ilp-quoter/index.js]: Yes - the ilp-quoter backend doesn't do any arithmetic -- it's up to the external component to compute the correct amounts. So strings are used at this point. Where are the docs for the connector component located?

Comment (emschwartz) [inline src/backends/ilp-quoter/index.js]: General documentation is lacking right now, but some comments in the code should do it for now

Comment (gip) [inline src/backends/ilp-quoter/index.js]: Basic docs added.

File: src/backends/ilp-quoter/index.js (hunk 2 above)
Comment (MatthewPhinney) [inline src/backends/ilp-quoter/index.js]: Maybe add a comment to the code for clarity? I find it hard to follow without the explanation. Also, I would argue for naming the currency variables:
```
const currencyA = pair[0].slice(0, 3)
const currencyB = pair[1].slice(0, 3)
return [currencyA, currencyB]
```
or something like that. Otherwise it's not clear to me what this method returns.

Comment (gip) [inline src/backends/ilp-quoter/index.js]: Sure. Done.

**Comment by MatthewPhinney** [comment]
LGTM

File: src/backends/ilp-quoter/index.js (hunk 2 above)
Comment (clark800) [inline src/backends/ilp-quoter/index.js]: I think it would be more clear to do something like `pair.split('@', 2)[1]` and have a separate method that parses the string format into a POJO along with a comment that shows an example like `USD@http://localhost:3000`

Comment (gip) [inline src/backends/ilp-quoter/index.js]: Sure - pull request welcome. Note that I expect the way to configure the (currency, ledger) to change in the near future so this code will be refactored then anyway.
